src/file.cr

```diff
@@ -380,6 +380,62 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Matches *path* against *pattern*.
+  #
+  # The pattern syntax is similar to shell filename globbing. It may contain the following metacharacters:
+  #
+  # * `*` matches an unlimited number of arbitrary characters.
+  #   * `"*"` matches all regular files.
+  #   * `"c*"` matches all files beginning with `c`.
+  #   * `"*c"` matches all files ending with `c`.
+  #   * `"*c*"` matches all files that have `c` in them (including at the beginning or end).
+  # * `?` matches any one charachter.
+  def self.fnmatch(pattern, path)
+    # linear-time algorithm adapted from https://research.swtch.com/glob
+    preader = Char::Reader.new(pattern)
+    sreader = Char::Reader.new(path)
+    next_ppos = 0
+    next_spos = 0
+    strlen = path.bytesize
+
+    while true
+      pnext? = preader.current_char != Char::ZERO
```

Comment:
`preader.has_next?`

Same in the next line

Also, `pnext?` as a variable name is an unfortunate accident of the current parser, that shouldn't be legal (at least it's not legal in Ruby). It's quite confusing. So it's better to use another name because that will break in an eventual change/fix.